ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/AtomicActionRecoveryModule.java

```diff
@@ -225,6 +227,20 @@ private void processTransactionsStatus()
                             _transactionType) != StateStatus.OS_UNKNOWN) {
                         doRecoverTransaction(currentUid);
                     }
+
+                    /*
+                     * If the current AtomicAction has been recovered,
+                     * its StateStatus should be OS_UNKNOWN.
+                     * If that is not the case, it means that the current
+                     * AtomicAction still needs to be recovered and this
+                     * SuspendBlockingRecoveryModule implementation should
+                     * block the suspension of the Recovery Manager
+                     */
+                    if (_recoveryStore.currentState(currentUid,
```

Comment:
`rcvAtomicAction.status()` returns the actionStatus which you can use to decide whether or not there is more work to be done on this action.

If the attempt to get the currentState throws an exception what should the value of hasWork be?

Reply:
> If the attempt to get the currentState throws an exception what should the value of hasWork be?

Do you think we should retry? I mean, in case an exception is thrown, should we set `hasWork = true` and retry at the next recovery cycle?

Reply:
I think we need to retry (i.e. set `hasWork` to `true`) because it would seem to be unsafe if we ignore failures in accessing the ObjectStore

Reply:
Thanks for your message, Tom. I will set `hasWork` to `true` in case of failures